### PR TITLE
Add vCenterTls Module

### PR DESCRIPTION
**Pull request by azirish1998**
Add module for managing vCenter certificates via CIS REST APIs

### CODE REVIEW

**Comment by kmruddy** [comment]
Module looks good, though I'm a bit confused. Is there a reason that you initialize the connection with the CIS module and then switch back to the REST calls? I would think it would be easier to continue forward with the CIS module. (Or skipping the usage of the CIS module and leave it completely in REST)

**Comment by azirish1998** [comment]
Thank you.
Nothing more than laziness I suppose :)  I guess a pure REST module would be cleaner, but I just didn't feel the need to write functions for (multiple) session management when connect-cisserver worked fine.

File: Modules/vCenterTls/vCenterTls.psm1 [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+                [System.Net.ServicePointManager]::CertificatePolicy = $null
+            }
+        }
+    }
+    End
+    {
+    }
+}
+
+function Invoke-CisRest {
```
Comment (dmilov) [inline Modules/vCenterTls/vCenterTls.psm1:619]: Rename to Invoke-VCRestWithCISSession, or at least document what exactly CisRest means

File: Modules/vCenterTls/vCenterTls.psm1 [hunk 2]
```diff
@@ -0,0 +1,691 @@
...
+
+    [CmdletBinding()]
+    Param
+    (
+        [Parameter()]
+        [VMware.VimAutomation.Cis.Core.Types.V1.CisServer]
+        $Server=$Global:DefaultCisServers[0],
```
Comment (dmilov) [inline Modules/vCenterTls/vCenterTls.psm1:626]: Add ValidateNotNull.
If function is called when no connection is available it should fail properly

File: Modules/vCenterTls/vCenterTls.psm1 [hunk 3]
```diff
@@ -0,0 +1,691 @@
...
+    
+        [Parameter(Mandatory=$true)]
+        [ValidateSet('Get','Post','Put','Delete','Patch')]
+        [string]
+        $Method,
```
Comment (dmilov) [inline Modules/vCenterTls/vCenterTls.psm1:631]: To be precise the type of this parameter should be [Microsoft.PowerShell.Commands.WebRequestMethod]

File: Modules/vCenterTls/vCenterTls.psm1 [hunk 4]
```diff
@@ -0,0 +1,691 @@
...
+    {
+        $Uri = 'https://'+$Server.Name+'/rest'+$UriPath
+        $Headers = @{'vmware-api-session-id' = $Server.SessionSecret
+                     'Accept' = 'application/json'
+                     }
+        $ContentType = 'application/json'
+
+        if($Body) {
+            # ConvertTo-Json escapes '\n' which we don't want.
+            $JsonBody = ($Body | ConvertTo-Json -Depth 10).Replace('\\n','\n')
```
Comment (dmilov) [inline Modules/vCenterTls/vCenterTls.psm1:654]: You can use -Compress switch instead of .Replace()

File: Modules/vCenterTls/vCenterTls.psm1 [hunk 5]
```diff
@@ -0,0 +1,691 @@
...
+        }
+        try {
+            switch($Method) {
+                'Get' {
+                    $Response = Invoke-RestMethod -Method $Method -Uri $Uri -Body $Body -Headers $Headers
+                    $Response.value
```
Comment (dmilov) [inline Modules/vCenterTls/vCenterTls.psm1:660]: Fix casing of 'Value' property

File: Modules/vCenterTls/vCenterTls.psm1 [hunk 6]
```diff
@@ -0,0 +1,691 @@
...
+                     'Accept' = 'application/json'
+                     }
+        $ContentType = 'application/json'
+
+        if($Body) {
+            # ConvertTo-Json escapes '\n' which we don't want.
+            $JsonBody = ($Body | ConvertTo-Json -Depth 10).Replace('\\n','\n')
+        }
+        try {
+            switch($Method) {
```
Comment (dmilov) [inline Modules/vCenterTls/vCenterTls.psm1:657]: You don't have to switch, return response when available instead

**Comment by azirish1998** [comment]
Thanks for the comments.  There is an error in the formatting of the payload for the REST call, so I'm going to close the pull request.

**Pull request closed without merging**